Library/Homebrew/dev-cmd/bump-formula-pr.rb

```diff
@@ -51,32 +51,37 @@ def bump_formula_pr
       safe_system "brew", "update"
     end
 
-    Utils::Inreplace.inreplace(formula.path) do |s|
-      if new_url_hash
-        old_url = formula_spec.url
-        if ARGV.dry_run?
-          ohai "replace '#{old_url}' with '#{new_url}'"
-          ohai "replace '#{old_hash}' with '#{new_hash}'"
-        else
-          s.gsub!(old_url, new_url)
-          s.gsub!(old_hash, new_hash)
-        end
-      else
-        resource_specs = formula_spec.specs
-        old_tag = resource_specs[:tag]
-        old_revision = resource_specs[:revision]
-        if ARGV.dry_run?
-          ohai "replace '#{old_tag}' with '#{new_tag}'"
-          ohai "replace '#{old_revision}' with '#{new_revision}'"
-        else
-          s.gsub!(/['"]#{old_tag}['"]/, "\"#{new_tag}\"")
-          s.gsub!(old_revision, new_revision)
-        end
-      end
+    s = (formula.path.open("r") { |f| Formulary.set_encoding(f).read }).extend(StringInreplaceExtension)
+    old_formula_version = Formulary.from_contents(formula.name, formula.path, s, requested_spec).version
+
+    if new_url_hash
+      old_url = formula_spec.url
+      ohai "replace '#{old_url}' with '#{new_url}'" if ARGV.dry_run?
+      s.gsub!(old_url, new_url)
+      ohai "replace '#{old_hash}' with '#{new_hash}'" if ARGV.dry_run?
+      s.gsub!(old_hash, new_hash)
+    else
+      resource_specs = formula_spec.specs
+      old_tag = resource_specs[:tag]
+      old_revision = resource_specs[:revision]
+      ohai "replace '#{old_tag}' with '#{new_tag}'" if ARGV.dry_run?
+      s.gsub!(/['"]#{old_tag}['"]/, "\"#{new_tag}\"")
+      ohai "replace '#{old_revision}' with '#{new_revision}'" if ARGV.dry_run?
+      s.gsub!(old_revision, new_revision)
+    end
+
+    raise Utils::InreplaceError.new({ formula.path => s.errors }) if s.errors.any?
+
+    new_formula_version = Formulary.from_contents(formula.name, formula.path, s, requested_spec).version
+
+    if old_formula_version == new_formula_version
```

Comment:
I'd probably make this check `old_formula_version >= new_formula_version` instead or handle that as two separate cases (`==` and `>`). This will avoid accidentally downgrading a formula, particularly in sometimes surprising cases like `Version.new("1.0.2a") < Version.new("1.0.2")`.